src/backends/beamcoder.ts

```diff
@@ -1,91 +1,39 @@
+import { BaseExtractor } from '../BaseExtractor';
+import type { Extractor, ExtractorArgs, InterpolateMode } from '../../framefusion';
+import { DownloadVideoURL } from '../DownloadVideoURL';
 
-/**
- * Assumptions made by this library:
- *
- *  - The input is always a mp4
- *    It would be nice to support other formats. One thing to change is the iformat, hardcoded for mp4.
- *
- */
-export const probeCodecPar = codecpar => ({
-    type: codecpar.type,
-    codec_type: codecpar.codec_type,
-    codec_id: codecpar.codec_id,
-    name: codecpar.name,
-    codec_tag: codecpar.codec_tag,
-    extradata: codecpar.extradata,
-    format: codecpar.format,
-    bit_rate: codecpar.bit_rate,
-    bits_per_coded_sample: codecpar.bits_per_coded_sample,
-    bits_per_raw_sample: codecpar.bits_per_raw_sample,
-    profile: codecpar.profile,
-    level: codecpar.level,
-    width: codecpar.width,
-    height: codecpar.height,
-    sample_aspect_ratio: codecpar.sample_aspect_ratio,
-    field_order: codecpar.field_order,
-    color_range: codecpar.color_range,
-    color_primaries: codecpar.color_primaries,
-    color_trc: codecpar.color_trc,
-    color_space: codecpar.color_space,
-    chroma_location: codecpar.chroma_location,
-    video_delay: codecpar.video_delay,
-    channel_layout: codecpar.channel_layout,
-    channels: codecpar.channels,
-    sample_rate: codecpar.sample_rate,
-    block_align: codecpar.block_align,
-    frame_size: codecpar.frame_size,
-    initial_padding: codecpar.initial_padding,
-    trailing_padding: codecpar.trailing_padding,
-    seek_preroll: codecpar.seek_preroll,
-});
-
-export const probeStream = stream => ({
-    type: stream.type,
-    index: stream.index,
-    id: stream.id,
-    time_base: stream.time_base,
-    start_time: stream.start_time,
-    duration: stream.duration,
-    nb_frames: stream.nb_frames,
-    disposition: stream.disposition,
-    discard: stream.discard,
-    sample_aspect_ratio: stream.sample_aspect_ratio,
-    metadata: stream.metadata,
-    avg_frame_rate: stream.avg_frame_rate,
-    attached_pic: stream.attached_pic,
-    side_data: stream.side_data,
-    event_flags: stream.event_flags,
-    r_frame_rate: stream.r_frame_rate,
-    codecpar: probeCodecPar(stream.codecpar),
-});
+const VERBOSE = false;
 
+const createDecoder = ({
+    demuxer,
+    streamIndex,
+    threadCount,
+}: {
+    demuxer: Demuxer;
+    streamIndex: number;
+    threadCount: number;
+}): Decoder => {
+    return beamcoder.decoder({
+        demuxer: demuxer,
+        width: demuxer.streams[streamIndex].codecpar.width,
+        height: demuxer.streams[streamIndex].codecpar.height,
+        stream_index: streamIndex,
+        pix_fmt: demuxer.streams[streamIndex].codecpar.format,
+        thread_count: threadCount,
+    });
+};
 
-/**
- * A filter to convert between color spaces.
- * An example would be YUV to RGB, for mp4 to png conversion.
- */
-const getFilter = async({
+const createFilter = async({
```

Comment:
to keep the previous comment

```suggestion
/**
 * A filter to convert between color spaces.
 * An example would be YUV to RGB, for mp4 to png conversion.
 */
const createFilter = async({
```